bundles/org.openhab.core.automation.module.script/src/main/java/org/openhab/core/automation/module/script/ScriptEngineFactory.java

```diff
@@ -58,6 +58,15 @@ public interface ScriptEngineFactory {
      */
     List<String> getScriptTypes();
 
+    /**
+     * This method returns whether the ScriptEngine supports pre-compilation of scripts.
+     *
+     * @return true if pre-compilation is supported, else false
+     */
+    default boolean supportsPreCompilation() {
```

Comment:
This would work as well, but then I would need to always initialize the script engine when the script action/condition is created, even if it does not implement Compilable. At the moment, the script engine is initialised on first use.

Reply:
Is that really an issue? If the engine implements `Compilable` there is no extra-cost. If it doesn't it just shifts the time when the engine is created from first usage to startup (which probably also improves performance when the script is actually called).

Reply:
That's a good argument.
I thought there might be a reason it was implemented this way, but creating the engine at start up actually seems like a good idea. This also avoids that the memory usage seems to constantly grow over time.

I'll implement that.